tests/src/roomHandlers/fishingRoomHandler.test.ts

```diff
@@ -0,0 +1,138 @@
+import { testSingletonModule } from '../../testHelpers/semanticMocha';
+import { FishingRoomHandler } from '../../../src/roomHandlers/fishingRoomHandler';
+import {
+  buildStateTransitionHelpers,
+  ArrangedTransitionData,
+} from '../../testHelpers/buildStateTransitionHelpers';
+import { RoomType } from '../../../src/utils/types';
+
+const roomType: RoomType = 'fishingRoom';
+type TRoomType = typeof roomType;
+type TArrangedTransitionData = ArrangedTransitionData<TRoomType>;
+
+testSingletonModule(
+  'roomHandlers/FishingRoomHandler',
+  ({ testIntegration }) => {
+    testIntegration('run', ({ testScenario }) => {
+      const {
+        testDefaultCommandAtEntrance,
+        testInvalidCommandAtEntrance,
+        testStateTransition,
+      } = buildStateTransitionHelpers(
+        testScenario,
+        roomType,
+        new FishingRoomHandler(),
+      );
+
+      testDefaultCommandAtEntrance();
+
+      testInvalidCommandAtEntrance();
+
+      testStateTransition(
+        'AtEntrance --> AtEntrance: fish',
+        (): TArrangedTransitionData => ({
+          startingRoomState: {
+            fishCaught: 0,
+            isRodBroken: true,
+            randomNumber: 0,
+          },
+          stateAssertionDescription: 'You can not fish with a broken rod',
+          expectedResult: {
+            commandDescription:
+              'You sit down and start to get ready to fish. But your rod is broken and you have to fix it.',
+            roomState: {
+              fishCaught: 0,
+              isRodBroken: true,
+              randomNumber: 0,
+            },
+          },
+        }),
+      );
+
+      testStateTransition(
+        'AtEntrance --> Fishing: fish',
+        (): TArrangedTransitionData => ({
+          startingRoomState: {
+            fishCaught: 0,
+            isRodBroken: false,
+            randomNumber: 0,
+          },
+          stateAssertionDescription: null,
+          expectedResult: {
+            commandDescription:
+              'You sit down and start to get ready to fish. This is going to be alot of fun!',
+            roomState: {
+              fishCaught: 0,
+              isRodBroken: false,
+              randomNumber: 0,
+            },
+          },
+        }),
+      );
+
+      testStateTransition(
+        'Fishing --> Fishing: fish',
+        (): TArrangedTransitionData => ({
+          startingRoomState: {
+            fishCaught: 0,
+            isRodBroken: false,
+            randomNumber: 0,
+          },
+          stateAssertionDescription:
+            'Did not catch any fish and your pole is still working',
+          expectedResult: {
+            commandDescription:
+              'You cast your fishing line. And nothing happen, maybe try again?',
+            roomState: {
+              fishCaught: 0,
+              isRodBroken: false,
+              randomNumber: 0,
```

Comment:
I'm running into the issue where `randomNumber` is... random after the command is ran. I'm not sure if I should update `testStateTransition` or stub this or something else ...?

Reply:
See point 3 in [this comment](https://github.com/normal-devs/klicker-knight/pull/44#discussion_r832232187).

Due to the complexity of your room you might have to update the test helper function.